pkg/cmd/server/api/validation/master.go

```diff
@@ -131,6 +135,36 @@ func ValidateProjectNodeSelector(nodeSelector string) fielderrors.ValidationErro
 	return allErrs
 }
 
+func ValidateServiceAccountConfig(config api.ServiceAccountConfig) fielderrors.ValidationErrorList {
+	allErrs := fielderrors.ValidationErrorList{}
+
+	for i, name := range config.ManagedNames {
+		if ok, msg := kvalidation.ValidateServiceAccountName(name, false); !ok {
+			allErrs = append(allErrs, fielderrors.NewFieldInvalid(fmt.Sprintf("", i), name, msg))
+		}
+	}
+
+	if len(config.PrivateKeyFile) > 0 {
+		if fileErrs := ValidateFile(config.PrivateKeyFile, "privateKeyFile"); len(fileErrs) > 0 {
+			allErrs = append(allErrs, fileErrs...)
+		} else if privateKey, err := serviceaccount.ReadPrivateKey(config.PrivateKeyFile); err != nil {
+			allErrs = append(allErrs, fielderrors.NewFieldInvalid("privateKeyFile", config.PrivateKeyFile, err.Error()))
+		} else if err := privateKey.Validate(); err != nil {
+			allErrs = append(allErrs, fielderrors.NewFieldInvalid("privateKeyFile", config.PrivateKeyFile, err.Error()))
+		}
+	}
+
+	for i, publicKeyFile := range config.PublicKeyFiles {
```

Comment:
It's time to add the concept of warnings.  We can't stop the server from starting (service accounts aren't THAT important), but we definitely want to notify if there are no public keys.